fenix/app/src/main/java/org/mozilla/fenix/extension/WebExtensionPromptFeature.kt

```diff
@@ -51,26 +55,47 @@ class WebExtensionPromptFeature(
             flow.mapNotNull { state ->
                 state.webExtensionPromptRequest
             }.distinctUntilChanged().collect { promptRequest ->
-                // The install flow in Fenix relies on an [Addon] object so let's convert the (GeckoView)
-                // extension into a minimal add-on. The missing metadata will be fetched when the user
-                // opens the add-ons manager.
-                val addon = Addon.newFromWebExtension(promptRequest.extension)
 
                 when (promptRequest) {
-                    is WebExtensionPromptRequest.Permissions -> handlePermissionRequest(
-                        addon,
-                        promptRequest,
-                    )
+                    is WebExtensionPromptRequest.AfterInstallation -> {
+                        handleAfterInstallationRequest(promptRequest)
+                    }
 
-                    is WebExtensionPromptRequest.PostInstallation -> handlePostInstallationRequest(
-                        addon.copy(installedState = promptRequest.extension.toInstalledState()),
-                    )
+                    is WebExtensionPromptRequest.BeforeInstallation.InstallationFailed -> {
+                        handleBeforeInstallationRequest(promptRequest)
+                        consumePromptRequest()
+                    }
                 }
             }
         }
         tryToReAttachButtonHandlersToPreviousDialog()
     }
 
+    private fun handleAfterInstallationRequest(promptRequest: WebExtensionPromptRequest.AfterInstallation) {
+        // The install flow in Fenix relies on an [Addon] object so let's convert the (GeckoView)
+        // extension into a minimal add-on. The missing metadata will be fetched when the user
+        // opens the add-ons manager.
+        val addon = Addon.newFromWebExtension(promptRequest.extension)
+        when (promptRequest) {
+            is WebExtensionPromptRequest.AfterInstallation.Permissions -> handlePermissionRequest(
+                addon,
+                promptRequest,
+            )
+            is WebExtensionPromptRequest.AfterInstallation.PostInstallation -> handlePostInstallationRequest(
+                addon.copy(installedState = promptRequest.extension.toInstalledState()),
+            )
+        }
+    }
+    private fun handleBeforeInstallationRequest(promptRequest: WebExtensionPromptRequest.BeforeInstallation) {
+        when (promptRequest) {
+            is WebExtensionPromptRequest.BeforeInstallation.InstallationFailed -> {
+                handleInstallationFailedRequest(
+                    exception = promptRequest.exception,
+                )
+                consumePromptRequest()
+            }
+        }
+    }
```

Comment:
```suggestion

        private fun handleBeforeInstallationRequest(promptRequest: WebExtensionPromptRequest.BeforeInstallation) {
        when (promptRequest) {
            is WebExtensionPromptRequest.BeforeInstallation.InstallationFailed -> {
                handleInstallationFailedRequest(
                    exception = promptRequest.exception,
                )
                consumePromptRequest()
            }
        }
    }
    
```

(blank lines)

Reply:
Now I realized why I was missing so many blank lines, my IDE was giving me the wrong impression of an empty line with the git author information 😅 . 
 
![image](https://github.com/mozilla-mobile/firefox-android/assets/773158/421e825e-a6ed-4d70-8a81-dde11bc2f236)

Reply:
Now it's accurate XD 
![image](https://github.com/mozilla-mobile/firefox-android/assets/773158/6f3320a6-bb90-4fda-9666-63c0c608824c)